### PR TITLE
Adds links from the Job views to quickly filter the Job Results

### LINKED ISSUE #6030: Add links from the Job views to quickly filter the Job Results
### As ...

Nelly - Network Engineer

### I want ...

To be able to start at the Job list view or the Job detail view and have a quick way to look at all Job Results for only that specific Job

### So that ...

I don't need to first go to the Job Result list view and then filter by Job, but instead can have a quick and seamless transition from finding the Job to seeing a list of all the Job Results for it.

### I know this is done when...

- I can click a single action button from the Job list view to get to the filtered list of Job Results for that Job
- I can see a count of Job Results that exists for a Job on its detail view page
- I can click on the count of Job Results on the detail view and it takes me to the filtered list of Job Results for that Job

### Optional - Feature groups this request pertains to.

- [ ] Automation
- [ ] Circuits
- [ ] DCIM
- [ ] IPAM
- [ ] Misc (including Data Sources)
- [ ] Organization
- [ ] Apps (and other Extensibility)
- [ ] Security (Secrets, etc)
- [ ] Image Management
- [X] UI/UX
- [ ] Documentation
- [ ] Other (not directly a platform feature)

### Database Changes

None

### External Dependencies

None

### PR DESCRIPTION
**Pull request by joewesch**
# Closes #6030
# What's Changed
This PR simply adds a few QOL buttons/links to quickly get from a Job to a filtered list of Job Results for that Job.

# Screenshots

**New Action button on Job list view:**

Screenshot 2024-08-01 at 12 34 55 PM

**New Job Result count and link on Job detail view:**

Screenshot 2024-08-01 at 12 35 28 PM

# TODO

- [x] Explanation of Change(s)
- [x] Added change log fragment(s) (for more information see [the documentation](https://docs.nautobot.com/projects/core/en/stable/development/#creating-changelog-fragments))
- [x] Attached Screenshots, Payload Example
- [ ] Unit, Integration Tests
- [ ] Documentation Updates (when adding/changing features)
- [ ] Example App Updates (when adding/changing features)
- [ ] Outline Remaining Work, Constraints from Design

### CODE REVIEW

File: nautobot/extras/tables.py [hunk 1]
```diff
@@ -1149,7 +1178,9 @@ class Meta(BaseTable.Meta):
 class AssociatedContactsTable(StatusTableMixin, RoleTableMixin, BaseTable):
     pk = ToggleColumn()
     contact_type = tables.TemplateColumn(
-        CONTACT_OR_TEAM_ICON, verbose_name="Type", attrs={"td": {"style": "width:20px;"}}
+        CONTACT_OR_TEAM_ICON,
+        verbose_name="Type",
+        attrs={"td": {"style": "width:20px;"}},
```
Comment (glennmatthews) [inline nautobot/extras/tables.py:1183]: Autoformatter gone wild?

**Review by HanlinMiao — APPROVED** [review]

File: nautobot/extras/templates/extras/job_detail.html [hunk 2]
```diff
@@ -115,6 +115,17 @@
                 <td>{{ object.enabled | render_boolean }}</td>
                 <td></td>
             </tr>
+            <tr>
+                <td>Job Results</td>
+                <td>
+                    {% if object.job_results.all %}
```
Comment (glennmatthews) [inline nautobot/extras/templates/extras/job_detail.html:121]:
```suggestion
                    {% if object.job_results.exists %}
```

**Review by glennmatthews — APPROVED** [review]
Great stuff! Arguably belongs in `next` rather than `develop` but I won't hold you to that.

File: nautobot/extras/tables.py (hunk 1 above)
Comment (joewesch) [reply nautobot/extras/tables.py:1183]: Yeah, ruff did all that.

File: nautobot/extras/templates/extras/job_detail.html (hunk 2 above)
Comment (joewesch) [reply nautobot/extras/templates/extras/job_detail.html]: Thanks, that was copy pasta from location children.

**Pull request merged**